dbms/src/Storages/DeltaMerge/File/DMFileReader.cpp

```diff
@@ -534,5 +540,52 @@ void DMFileReader::readFromDisk(
     }
 }
 
+void DMFileReader::readColumn(ColumnDefine & column_define,
+                              ColumnPtr & column,
+                              size_t start_pack_id,
+                              size_t pack_count,
+                              size_t read_rows,
+                              size_t skip_packs,
+                              [[maybe_unused]] bool force_seek)
+{
+    if (!getCachedPacks(column_define.id, start_pack_id, pack_count, read_rows, column))
+    {
+        auto data_type = dmfile->getColumnStat(column_define.id).type;
+        auto col = data_type->createColumn();
+        readFromDisk(column_define, col, start_pack_id, read_rows, skip_packs, true /*force_seek*/);
```

Comment:
Why here always set `force_seek` to true?

Reply:
If block is read from cache, we don't need to read from file and the file offset is not changed.

We can trace whether read from cache of the previous read to optimize this case.

Reply:
I think we should not always set `force_seek = true`. It is a regression.

Reply:
Add `std::unordered_map<ColId, bool> last_read_from_cache;` to trace whether last read is from cache to avoid seek every time.